src/Microsoft.OData.Core/MultipartMixed/ODataMultipartMixedBatchWriter.cs

```diff
@@ -0,0 +1,413 @@
+        /// Flush the output.
+        /// </summary>
+        protected override void FlushSynchronously()
+        {
+            this.RawOutputContext.Flush();
+        }
+
+#if PORTABLELIB
+        /// <summary>
+        /// Flush the output.
+        /// </summary>
+        /// <returns>Task representing the pending flush operation.</returns>
+        protected override Task FlushAsynchronously()
+        {
+            return this.RawOutputContext.FlushAsync();
+        }
+#endif
+
+        /// <summary>
+        /// Starts a new changeset - implementation of the actual functionality.
+        /// </summary>
+        /// <param name="changeSetGuid">The value for changeset boundary for multipart batch.</param>
+        protected override void WriteStartChangesetImplementation(string changeSetGuid)
+        {
+            Debug.Assert(changeSetGuid != null, "changeSetGuid != null");
+
+            // write pending message data (headers, response line) for a previously unclosed message/request
+            this.WritePendingMessageData(true);
+
+            this.SetState(BatchWriterState.ChangesetStarted);
+            Debug.Assert(this.changeSetBoundary == null, "this.changeSetBoundary == null");
+            this.changeSetBoundary = ODataMultipartMixedBatchWriterUtils.CreateChangeSetBoundary(this.RawOutputContext.WritingResponse, changeSetGuid);
+
+            // write the boundary string
+            ODataMultipartMixedBatchWriterUtils.WriteStartBoundary(this.RawOutputContext.TextWriter, this.batchBoundary, !this.batchStartBoundaryWritten);
+            this.batchStartBoundaryWritten = true;
+
+            // write the change set headers
+            ODataMultipartMixedBatchWriterUtils.WriteChangeSetPreamble(this.RawOutputContext.TextWriter, this.changeSetBoundary);
+            this.changesetStartBoundaryWritten = false;
+        }
+
+        /// <summary>
+        /// Creates an <see cref="ODataBatchOperationRequestMessage"/> for writing an operation of a batch request
+        /// - implementation of the actual functionality.
+        /// </summary>
+        /// <param name="method">The Http method to be used for this request operation.</param>
+        /// <param name="uri">The Uri to be used for this request operation.</param>
+        /// <param name="contentId">The Content-ID value to write in ChangeSet head.</param>
+        /// <param name="payloadUriOption">
+        /// The format of operation Request-URI, which could be AbsoluteUri, AbsoluteResourcePathAndHost, or RelativeResourcePath.</param>
+        /// <param name="dependsOnIds">The prerequisite request ids of this request.</param>
+        /// <returns>The message that can be used to write the request operation.</returns>
+        protected override ODataBatchOperationRequestMessage CreateOperationRequestMessageImplementation(
+            string method, Uri uri, string contentId, BatchPayloadUriOption payloadUriOption,
+            IList<string> dependsOnIds)

```

Comment:
>IList<string> [](start = 12, length = 13)

Should be IEnumerable<string> rather than IList<string> (here and in any other input parameters, unless we are explicitly using the list (storing, adding/removing, etc.) which we generally shouldn't)